packages/backend/src/accounting/psql/ledger-transfer/index.ts

```diff
@@ -123,65 +123,67 @@ function validateTransferStateUpdate(
 
 export async function createTransfers(
   deps: ServiceDependencies,
-  transfers: CreateTransferArgs[],
-  trx?: TransactionOrKnex
+  transfers: CreateTransferArgs[]
 ): Promise<CreateTransfersResult> {
-  try {
-    const validationResults = await Promise.all(
-      transfers.map((transfer) =>
-        validateTransfer(deps, {
-          creditAccount: transfer.creditAccount,
-          debitAccount: transfer.debitAccount,
-          amount: transfer.amount,
-          transferRef: transfer.transferRef,
-          timeoutMs: transfer.timeoutMs
-        })
-      )
-    )
+  const trx = await deps.knex.transaction()
 
-    const errors: CreateTransferError[] = []
+  const errors: CreateTransferError[] = []
+  const results: LedgerTransfer[] = []
 
-    for (const [i, maybeError] of validationResults.entries()) {
-      if (isTransferError(maybeError)) {
-        errors.push({
-          index: i,
-          error: maybeError
-        })
-      }
+  for (const [index, transfer] of transfers.entries()) {
+    const error = await validateTransfer(deps, transfer, trx)
+
+    if (error) {
+      errors.push({ index, error })
+      continue
```

Comment:
or fail fast and
```suggestion
      break
```

Reply:
Maybe fail fast is better, you just lose visibility into each individual transfer (but maybe thats ok)

Reply:
Stared a psql follow-up issue: https://github.com/interledger/rafiki/issues/1137